proxy.c

```diff
@@ -265,9 +265,9 @@ void forward_data(int source_sock, int destination_sock) {
         exit(EXIT_FAILURE);
     }
 
-    while ((n = splice(source_sock, NULL, buf_pipe[WRITE], NULL, BUF_SIZE, SPLICE_F_MOVE)) > 0) {
-        if (splice(buf_pipe[READ], NULL, destination_sock, NULL, n, SPLICE_F_MOVE) < 0) {
-            perror("splice");
+    while ((n = splice(source_sock, NULL, buf_pipe[WRITE], NULL, SSIZE_MAX, SPLICE_F_NONBLOCK|SPLICE_F_MOVE)) > 0) {
+        if (splice(buf_pipe[READ], NULL, destination_sock, NULL, SSIZE_MAX, SPLICE_F_MOVE) < 0) {
+            syslog(LOG_ERR, "write splice: %m");
```

Comment:
I missed your second reply, if you'd like, I can also create a pull request for this, I already have the code written in my branch.